packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { Text } from '@deriv/components';
+import { Localize, localize } from '@deriv/translations';
+import { Wizard } from '@deriv/ui';
+
+type TSignupWizardProps = {
+    setIsWizardOpen: (is_wizard_open: boolean) => void;
+};
+
+const SignupWizard = ({ setIsWizardOpen }: TSignupWizardProps) => {
+    const [create_app_state, setCreateAppState] = React.useState({});
+    const [current_step_key, setCurrentStepKey] = React.useState<string>();
+
+    const is_final_step = current_step_key === 'complete_step';
+
+    const closeWizardHandler = () => {
+        setIsWizardOpen(false);
+    };
+
+    const handleComplete = () => {
+        setIsWizardOpen(false);
+    };
+
+    const onChangeStep = (_current_step: number, _current_step_key?: string) => {
+        setCurrentStepKey(_current_step_key);
+    };
+
+    return (
+        <Wizard
+            has_dark_background
+            lock_final_step={false}
+            onClose={closeWizardHandler}
+            onComplete={handleComplete}
+            wizard_title={localize('Become a payment agent')}
+            primary_button_label={is_final_step ? localize('Submit') : localize('Next')}
```

Comment:
```suggestion
            primary_button_label={current_step === 3 ? localize('Submit') : localize('Next')}
```
@george-usynin-binary Couldn’t we just simply check for the last index? 🤔

Reply:
@farzin-deriv some steps can be hidden and the last index will not be the last step, using step_key is mentioned in library(documentation/usage) itself, that's why i use step_key